src/psyclone/tests/psyir/symbols/datatype_test.py

```diff
@@ -79,6 +79,12 @@ def test_unresolvedtype_str():
     assert str(data_type) == "UnresolvedType"
 
 
+def test_unresolvedtype_is_allocatable():
+    '''Test that the UnresolvedType class' is_allocatable property works.'''
+    data_type = UnresolvedType()
+    assert data_type.is_allocatable is False
```

Comment:
If it's unresolved, we don't know whether it's allocatable so this should be `None`?

Reply:
Good point. I was originally thinking of even making the default (in `Datatype`) to return None, but then we get all scalars returned as unknown, unless I'll fix that as well. I left that with a TODO :) So, UnresolvedType now returns None.